scene/3d/physics_body_3d.cpp

```diff
@@ -1130,39 +1128,37 @@ bool CharacterBody3D::move_and_slide() {
 
 	for (int iteration = 0; iteration < max_slides; ++iteration) {
 		PhysicsServer3D::MotionResult result;
-		bool found_collision = false;
-
 		bool collided = move_and_collide(motion, result, margin, false, !sliding_enabled);
 		if (!collided) {
```

Comment:
You can remove that condition, only need to do `if(collided)` because at the end the condition check if `collided`